### PR TITLE
[performance_optim] reduce frequency of declaring attention_mask in Ascend NPU flash attention

### PR DESCRIPTION
**Pull request by FightingZhen**
# What does this PR do?

During analyzing performance of flash attention on Ascend NPU, we find that a fixed-size 2048*2048 upper triangular matrix `attn_mask_npu` is frequently defined in func `npu_flash_attn_func` and `npu_flash_attn_varlen_func`, which will cost a lot of time.

This PR is committed for solving this problem, by defining and using global variable `ATTN_MASK_NPU`, this matrix will not be defined every time func `npu_flash_attn_func` or `npu_flash_attn_varlen_func` is called.

Thanks @CurryRice233 for proposing this idea.

Fixes # (issue)
Not related.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by SunMarc — COMMENTED** [review]
Thanks ! what is the increase of speed with this change ?

**Comment by FightingZhen** [comment]
> Thanks ! what is the increase of speed with this change ?

Yes, this modification can increase the speed

**Comment by CurryRice233** [comment]
> Thanks ! what is the increase of speed with this change ?

With this change, I don't have to re-create a mask every FA, just use one created, which reduces the creation time and memory of the mask for backward.

**Comment by FightingZhen** [comment]
@ArthurZucker please help me review and merge this PR, thanks a lot :)

**Review by ArthurZucker — APPROVED** [review]
Sure, not sure I understand why this alwasys works, when the inputs can be longer but let's merge
